code/datums/gamemode/role/role.dm

```diff
@@ -124,7 +124,10 @@
 	M.antag_roles.Add(id)
 	M.antag_roles[id] = src
 
-	OnPreSetup()
+	if (!OnPreSetup())
+		return FALSE
+	if (!OnPostSetup())
```

Comment:
Please do not call OnPostSetup here. It's to be called after the game has been set up, and AssignToRole's called before that.